library/std/src/sys/pal/unix/fs.rs

```diff
@@ -1574,6 +1616,12 @@ impl fmt::Debug for File {
     }
 }
 
+impl fmt::Debug for Mode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "0o{:03o}", self.0)
```

Comment:
I am not sure if this is the ideal formatting, but it should be 4-padded at least. It is usually written padded to the 4th digit, so that the suid, guid, and sticky bits don't change the offset.
```suggestion
        write!(f, "0o{:04o}", self.0)
```

Reply:
I ended up going with `{:06o}` because libc::S_IFMT is 0o170000 i.e. there are some higher bits that determine the file type. https://www.gnu.org/software/libc/manual/html_node/Testing-File-Type.html

Here is a related PR you might have feedback on: https://github.com/rust-lang/libc/pull/3634.